### PR TITLE
Build minified for production

### PR DESCRIPTION
**Pull request by jerome-benoit**
The gain in size if around 50%

Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

## PR Checklist

- [x] Please add a description of your changes.
- [ ] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

### CODE REVIEW

**Comment by pioardi** [comment]
Please hold on I see that coverage Is at 0 on this branch
Tomorrow I Will look into that, I have also one question.
This Will minify code only when we release?

**Comment by jerome-benoit** [comment]
> Please hold on I see that coverage Is at 0 on this branch
> Tomorrow I Will look into that, I have also one question.

Not sure it's related to that PR.

> This Will minify code only when we release?

Yes or if you do not use *:dev scripts for testing.

**Comment by pioardi** [comment]
> > Please hold on I see that coverage Is at 0 on this branch
> > Tomorrow I Will look into that, I have also one question.
> 
> Not sure it's related to that PR.
> 
> > This Will minify code only when we release?
> 
> Yes or if you do not use *:dev scripts for testing.


Yes I Need to look into that Better because on master branch I see the coverage all tight, so I Will look into this Better tommorow.

About the changes I was just thinking that se can stay with the current commands and just add the others for prof, something Like build:prod, test:prod and so on. In this way we sinplify script naming for developer that are used much more often.

What are your thoughts on that?

**Comment by pioardi** [comment]
I think the coverage on sonar is not reported with those changes.
I see that for other branches coverage is reported ( https://sonarcloud.io/dashboard?id=pioardi_poolifier&pullRequest=153 )

**Comment by pioardi** [comment]
This is the main issue, lcov.info file previously had path with lib/ and now with src/...
mmmmmm we can try to change sonar.scm.exclusions.disabled=false ( was true) into our sonar-project.properties file


image

**Comment by jerome-benoit** [comment]
> I think the coverage on sonar is not reported with those changes.
> I see that for other branches coverage is reported ( [sonarcloud.io/dashboard?id=pioardi_poolifier&pullRequest=153](https://sonarcloud.io/dashboard?id=pioardi_poolifier&pullRequest=153) )

There's no changes in that PR that change the CI behaviour: same build is used except for npm publication. 
If it fails, that's not because of that PR.

**Comment by pioardi** [comment]
> > I think the coverage on sonar is not reported with those changes.
> > I see that for other branches coverage is reported ( [sonarcloud.io/dashboard?id=pioardi_poolifier&pullRequest=153](https://sonarcloud.io/dashboard?id=pioardi_poolifier&pullRequest=153) )
> 
> There's no changes in that PR that change the CI behaviour: same build is used except for npm publication.
> If it fails, that's not because of that PR.

I am not sure jerome, I see that for another PR open this morning it works well ( https://sonarcloud.io/dashboard?id=pioardi_poolifier&pullRequest=157 )

**Comment by pioardi** [comment]
I think that the code it seems that code is minified during the dev build mmm
I saw those outputs only when I run npm run test:dev on this branch

image

**Comment by jerome-benoit** [comment]
> I think that the code it seems that code is minified during the dev build mmm
> I saw those outputs only when I run npm run test:dev on this branch
> 
> image

I can remove that output for CI.

**Comment by pioardi** [comment]
No I mean the problem is not the output itself but it could be related to the coverage problem maybe something has changed since that we do not see this output on other branches?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,14 +34,14 @@ jobs:
         run: npm ci
 
       - name: Build
-        run: npm run build --if-present
+        run: npm run build:dev --if-present
```
Comment (pioardi) [inline .github/workflows/ci.yml:37]: we can stay with the current commands and just add the others for prod, something Like build:prod, test:prod and so on. In this way we simplify script naming for developer that are used much more often.

Comment (pioardi) [reply .github/workflows/ci.yml:37]: And we also do not need to update the contributing.md

**Review by pioardi — COMMENTED** [review]
Changes are ok, I just have two main concerns:
  - Sonar coverage not working
  - Should we then run npm run test:prod periodically at least?

**Comment by jerome-benoit** [comment]
> Changes are ok, I just have two main concerns:
> 
> * Sonar coverage not working

I will comment out the build analysis code.

> * Should we then run npm run test:prod periodically at least?

No.

**Comment by pioardi** [comment]
@jerome-benoit  I will take a look on the coverage as soon as I can so that we can merge this

**Comment by jerome-benoit** [comment]
> @jerome-benoit I will take a look on the coverage as soon as I can so that we can merge this

The build command and the build output is now the same as before, so if that still does not work, that PR have nothing to do with it.

**Comment by pioardi** [comment]
> > @jerome-benoit I will take a look on the coverage as soon as I can so that we can merge this
> 
> The build command and the build output is now the same as before, so if that still does not work, that PR have nothing to do with it.

I agree with you command are now the same but this is the only branch where coverage is not reported so....

**Comment by jerome-benoit** [comment]
> > > @jerome-benoit I will take a look on the coverage as soon as I can so that we can merge this
> > 
> > 
> > The build command and the build output is now the same as before, so if that still does not work, that PR have nothing to do with it.
> 
> I agree with you command are now the same but this is the only branch where coverage is not reported so....

I've reverted to the exact previous rollup configuration and sonar still have the issue. So like I said, that PR have nothing to do with sonar showing 
—
Coverage
0.0%
Estimated after merge

**Comment by pioardi** [comment]
@jerome-benoit  it now works, take a look https://sonarcloud.io/dashboard?id=pioardi_poolifier&pullRequest=155

I just changed the sonar.sources property from lib to src, coverage decreased from 89 to 88 but I don't care we will need anyway to go through that

**Review by pioardi — APPROVED** [review]
@jerome-benoit  thanks to work on that, at you the honor to merge.

@Shinigami92 @jerome-benoit I almost done for today, will take a look again this evening see you

**Comment by jerome-benoit** [comment]
> @jerome-benoit it now works, take a look [sonarcloud.io/dashboard?id=pioardi_poolifier&pullRequest=155](https://sonarcloud.io/dashboard?id=pioardi_poolifier&pullRequest=155)
> 
> I just changed the sonar.sources property from lib to src, coverage decreased from 89 to 88 but I don't care we will need anyway to go through that

Well, the previous configuration was not correct then ;)
You only what to monitor src? 
Because other code like test might also be interesting to static analyse.

**Pull request merged**